org.eclipse.sisu.plexus.tests/src/org/eclipse/sisu/plexus/BeanConstantTest.java

```diff
@@ -194,6 +195,11 @@ public void testSimpleFileBean()
         assertEquals( "readme.txt", injector.getInstance( Key.get( File.class, Names.named( "README" ) ) ).getName() );
     }
 
+    public void testSimplePathBean()
+    {
+        assertEquals( "readme.txt", injector.getInstance( Key.get( Path.class, Names.named( "README" ) ) ).getFileName().toString() );
```

Comment:
why are the DEFAULT_CONVERTERS not used in this context?

Reply:
The original goal of this test was to check that `PlexusXmlBeanConverter` correctly fell-back to use the `TypeConverter`s registered with Guice. The full blown XML configurator API which includes `DEFAULT_CONVERTERS` was covered by other tests outside of this codebase.

Side note: the Plexus compatibility layer was developed using a "clean-room" approach which matched behaviour without looking at the original source. The tests to drive that were the Maven integration test-suite as well as other Plexus-based test suites. Unfortunately there was no single test suite that captured all of the historical Plexus behaviour, or exercised the whole of the API. We also didn't have the capacity at the time to (re)write all of those tests for this project, so each release is still run against those external test-suites. It would be great to build up the local tests to cover more of this historical API.

Getting back to the issue, it looks like there's a bug in this test where it doesn't wrap the test configuration to activate the additional converters registered in https://github.com/eclipse/sisu.inject/blob/master/org.eclipse.sisu.inject/src/org/eclipse/sisu/wire/WireModule.java#L32  - the `File` case just happened to work because `PlexusXmlBeanConverter`s last resort would be to try and use a constructor which accepts a `String`. I'll update the test to reflect how `PlexusXmlBeanConverter` would be configured inside the Plexus compatibility layer, which will fix the `Path` case.

Reply:
I removed the added test method from this PR as it is not related to the newly added converter. Feel free to readd in the context of another commit.